pkg/gitparse/gitparse.go

```diff
@@ -42,10 +44,45 @@ type Commit struct {
 type Diff struct {
 	PathB     string
 	LineStart int
-	Content   bytes.Buffer
-	IsBinary  bool
+	// Used to keep small diffs in memory and larger diffs in a file.
+	contentWriter *bufferedfilewriter.BufferedFileWriter
+	IsBinary      bool
+}
+
+type diffOption func(*Diff)
+
+// withPathB sets the PathB option.
+func withPathB(pathB string) diffOption { return func(d *Diff) { d.PathB = pathB } }
+
+// NewDiff creates a new Diff with a threshold.
+func NewDiff(opts ...diffOption) *Diff {
+	const defaultThreshold = 20 * 1024 * 1024 // 20MB
+	d := new(Diff)
+	for _, opt := range opts {
+		opt(d)
+	}
+	d.contentWriter = bufferedfilewriter.New(bufferedfilewriter.WithThreshold(defaultThreshold))
+
+	return d
+}
+
+// Len returns the length of the storage.
+func (d *Diff) Len() int { return d.contentWriter.Len() }
```

Comment:
These methods probably can be removed, but I was trying to avoid reaching into the contentWriter directly. This is also exported since we need access to this in the git source. Open to changing it if we feel these are providing any additional encapsulation.

Reply:
Do you want to guard them behind the state like `ReadCloser()`? I find it a bit surprising that some of the read methods are guarded and some aren't.

Reply:
I like that idea. Will guard them.